bindings/node/lib/clientEncryption.js

```diff
@@ -549,6 +559,69 @@ module.exports = function (modules) {
       return value;
     }
 
+    /**
+     * Creates a collection that has encrypted document fields
+     *
+     * @template {TSchema} - Schema for the collection being created
+     * @param {Db} db - A database to create the collection in
+     * @param {string} name - The name of the collection to be created
+     * @param {object} options - Options for createDataKey and for createCollection.
+     * @param {string} options.provider - provider name
+     * @param {ClientEncryptionCreateDataKeyProviderOptions} options.createDataKeyOptions - options to pass to createDataKey
+     * @param {CreateCollectionOptions} options.createCollectionOptions - options to pass to createCollection, must include `encryptedFields`
+     * @returns {Promise<{ collection: Collection<TSchema>, encryptedFields: Document }>} - created collection and generated encryptedFields
+     * @throws {MongoCryptCreateDataKeyForEncryptedCollectionError} - If part way through the process a createDataKey invocation fails, an error will be rejected that has the partial `encryptedFields` that were created.
+     * @throws {MongoCryptCreateEncryptedCollectionError} - If creating the collection fails, an error will be rejected that has the entire `encryptedFields` that were created.
+     */
+    async createEncryptedCollection(db, name, options) {
+      const {
+        provider,
+        createDataKeyOptions = {},
+        createCollectionOptions: {
+          encryptedFields: { ...encryptedFields },
+          ...createCollectionOptions
+        }
+      } = options;
+
+      if (createDataKeyOptions.keyAltNames != null) {
+        throw new MongoCryptInvalidArgumentError(
+          'Cannot pass createDataKeyOptions with keyAltNames set'
+        );
+      }
```

Comment:
It feels inconsistent to throw a type error for missing parameters in the destructuring and a `MongoCryptInvalidArgumentError` specifically for `keyAltNames`.  Now that we have `MongoCryptInvalidArgumentError`, we should use it for all invalid arguments.

Reply:
MongoCryptInvalidArgumentError is removed